std-bits/table/src/main/java/org/enso/table/aggregations/MinOrMax.java

```diff
@@ -0,0 +1,67 @@
+package org.enso.table.aggregations;
+
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.table.Column;
+import org.enso.table.data.table.problems.InvalidAggregation;
+
+import java.util.List;
+
+/***
+ * Aggregate Column finding the minimum or maximum entry in a group.
+ */
+public class MinOrMax extends AggregateColumn{
+  private final Storage storage;
+  private final int minOrMax;
+
+  public MinOrMax(String name, Column column, int minOrMax) {
+    super(name, Storage.Type.OBJECT);
+    this.storage = column.getStorage();
+    this.minOrMax = minOrMax;
+  }
+
+  @Override
+  public Object aggregate(List<Integer> indexes) {
+    Object current = null;
+    for (int row: indexes) {
+      Object value = storage.getItemBoxed(row);
+      if (value != null) {
+        try {
+          if (current == null || Compare(current, value) == minOrMax) {
```

Comment:
Would be good to explain what values `minOrMax` takes, as it is not immediately clear.

Also, I think Java does only require `compareTo` (which we delegate to) to return some positive or negative value - but it does not specify them to be `-1` and `1` and I'm pretty sure there are cases when this can indeed be a different value - in such cases this computation will be incorrect. I think it should be changed to check for the sign.